### PR TITLE
Add l431

### PR DESCRIPTION
**Pull request by Netzpfuscher**

### CODE REVIEW

**Comment by Netzpfuscher** [comment]
Added dq type for cleaner code
Fixed F401 mp2 hw conf

File: MESC_Common/Src/MESCcli.c [hunk 1]
```diff
@@ -896,7 +896,7 @@ static void cli_reply_begin( void )
     cli_reply_buffer_offset = 0;
 }
 
-static void cli_reply_close( void )
+void cli_reply_close( void )
```
Comment (cod3b453) [inline MESC_Common/Src/MESCcli.c:899]: Not clear why this has been changed?

File: MESC_Common/Src/MESCuart.c [hunk 2]
```diff
@@ -34,18 +34,29 @@
 #include "MESCfoc.h"
 #include "MESCmotor_state.h"
 
-extern UART_HandleTypeDef huart3;
+#ifdef USB
+#include "usbd_cdc_if.h"
+#endif
+
```
Comment (cod3b453) [inline MESC_Common/Src/MESCuart.c:40]: maybe match definition with implementation:
```
#else 
extern UART_HandleTypeDef HW_UART;
#endif
```

File: MESC_Common/Src/MESCuart.c [hunk 3]
```diff
@@ -34,18 +34,29 @@
 #include "MESCfoc.h"
 #include "MESCmotor_state.h"
 
-extern UART_HandleTypeDef huart3;
+#ifdef USB
```
Comment (cod3b453) [inline MESC_Common/Src/MESCuart.c:37]: Maybe a more verbose define such as `MESC_UART_USB`?

**Review by cod3b453 — APPROVED** [review]
Few minor things I picked up but nothing blocker

File: MESC_Common/Src/MESCcli.c (hunk 1 above)
Comment (Netzpfuscher) [reply MESC_Common/Src/MESCcli.c:899]: Need to change it back. Something I did during debug.

File: MESC_Common/Src/MESCuart.c (hunk 3 above)
Comment (Netzpfuscher) [reply MESC_Common/Src/MESCuart.c:37]: You are right. For the future we should make it possible to define multiple connections. That would need a change to the MESCcli to give the functions a cli handle or so

**Pull request merged**